### PR TITLE
WhiteBIT Exchange Extension

### PR DESCRIPTION
**Pull request by marco-whitebit**
## Description
WhiteBIT exchange extension for Raycast which uses private API for transfers, code applying, etc and public API for displaying market activity.  

## Screencast
Screencast

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Hi @marco-whitebit! 👋 

Is there a way to test this extension? Do you have any demo credentials? 🙂

**Comment by marco-whitebit** [comment]
Hello, you can create account using any email, no KYC required to test extension, you can claim demo token to test Codes or balances. 

After registration you can go to this page and claim it https://whitebit.com/codes/all?demo=true.

Let me know if it makes sense

**Comment by pernielsentikaer** [comment]
I think it does, I got the token from here, I used the first one

Raycast-rpeFRtdl@2x

And then I got the secret key from here

Raycast-pPeBKGio@2x

but it seems like it's not working as expected

whitebit 2023-10-11 at 10 34 43

Also, we should aim to die gracefully instead of crashing 🙂

Are you working at whitebit, feel free to DM me to help 🙂

File: extensions/whitebit/package.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    {
+      "name": "token",
+      "title": "Token",
+      "description": "Your API token.",
+      "type": "textfield",
+      "required": true
+    },
+    {
+      "name": "secret",
+      "title": "Sercey Key",
```
Comment (pernielsentikaer) [inline extensions/whitebit/package.json:112]:
```suggestion
      "title": "Sercet Key",
```

**Comment by marco-whitebit** [comment]
Thank you for your review, will add try..catch for HTTP exception. 

Can you please confirm that you enabled all endpoints described in README file and put the secret key and public key correctly? Currently it's a little bit confusing on our UI, firstly you will see a secret key and second is public, we gonna change the order of fields soon to make it more friendly :)

**Comment by marco-whitebit** [comment]
@pernielsentikaer sure, i will DM you in slack asap

**Comment by marco-whitebit** [comment]
@pernielsentikaer i sent DM in Slack

**Comment by pernielsentikaer** [comment]
Still struggling here, DM'ed you 🙂

**Comment by marco-whitebit** [comment]
Hello, @pernielsentikaer, added `catch` to avoid throwing errors and handle it with alert.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Apply Codes command should probably be a no-view command instead

#### General
 - Lists should have an `isLoading` property, maybe you can use [useCachedPromise](https://developers.raycast.com/utilities/react-hooks/usecachedpromise) - this also gives you {`data`, `isLoading`, `error`} out of the box
 - Don't change the navigationTitle in the root command - it will be automatically set to the command name. Use navigationTitle only in nested screens to provide additional context.

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by marco-whitebit** [comment]
Hello, @pernielsentikaer. Thank you for review, i made all requested changes.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**